splink/comparison_level_composition.py

```diff
@@ -178,6 +178,9 @@ def cl_not(
     cld = dicts[0]
     result["sql_condition"] = f"NOT ({cld['sql_condition']})"
 
+    if cld.get("is_null_level", False):  # invert if is_null_level
+        result["is_null_level"] = None
```

Comment:
Should assign False, not None? Maybe I don't understand

Reply:
We should really pop it instead, I believe.

Reply:
Maybe? That doesn't make sense to me immediately but I'll see what you come up with